### PR TITLE
refactor: reduce build time

### LINKED ISSUE #699: support to cache to old versions' documents
## Background

The current full compilation of the documentation for each project is performed each time a PR merge and scheduled deployment is performed. The older versions of these documents are barely modified, and at the same time, as the versions continue to iterate, the version numbers of the documents become more and more numerous, making the full compilation take longer and longer.

## Suggestion

Add a caching mechanism for older versions of documents so that we don't have to do a full compilation every time.

## Available options
- [ ] GitHub Workflows cache

### PR DESCRIPTION
**Pull request by SkyeYoung**
Fixes: #699 

Changes:

- reduce build time about 60%

Screenshots of the change:

none

### CODE REVIEW

**Comment by SkyeYoung** [comment]
@juzhiyuan @bzp2010 full build preview: https://625ab55e943e5306140b300e--full-build-apache-apisix.netlify.app/

**Comment by juzhiyuan** [comment]
origin_img_v2_52a7a4ee-66ba-4ce6-a812-23432f9eaf1g

LGTM!

File: website/config/apisix-versions.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  '2.13',
+];
+
+/**
+ * @type {Array<{label: string, href: string}>}
+ */
+const archivedVersions = [
+  {
+    label: '2.11',
+    href: 'https://625a9090d04b9a6953165811--2-11-old-docs-apache-apisix.netlify.app/docs/apisix/getting-started/',
```
Comment (juzhiyuan) [inline website/config/apisix-versions.js:17]: Hi, may I know how to get those links?

**Review by juzhiyuan — APPROVED** [review]
LGTM

File: website/config/apisix-versions.js (hunk 1 above)
Comment (SkyeYoung) [reply website/config/apisix-versions.js:17]: The site name `2-11-old-docs-apache-apisix.netlify.app` is entered manually. But the `625a9090d04b9a6953165811` is randomly generated. The full host will be returned by netlify cli. 

If the prod attribute is passed in during deployment, it seems that there will be no randomly generated ID. But I haven't tried yet.

**Review by Baoyuantop — APPROVED** [review]

File: package.json [hunk 2]
```diff
@@ -36,4 +37,4 @@
     "*.{yml,yaml}": "eslint --cache --fix",
     "*.css": "stylelint --cache --fix"
   }
-}
+}
```
Comment (bzp2010) [inline package.json:40]: EOL

**Review by bzp2010 — APPROVED** [review]

**Pull request merged**